controller.go

```diff
@@ -82,9 +82,10 @@ func (fn HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		status := http.StatusInternalServerError
 		var e *AppError
 		if errors.As(err, &e) {
-			status = e.StatusCode
+			replyJSON(w, e.JSONResponse(), e.StatusCode)
+		} else {
+			http.Error(w, err.Error(), status)
```

Comment:
this line could leak internal information as we do not enforce lower layers to return AppError.

Reply:
done